src/ast/passes/semantic_analyser.cpp

```diff
@@ -2034,36 +2038,83 @@ If you're seeing errors, try clamping the string sizes. For example:
   }
 }
 
-void SemanticAnalyser::visit(Sizeof &szof)
+std::optional<size_t> SemanticAnalyser::check(Sizeof &szof)
```

Comment:
nit on this name - how about `get_size` and then `get_offset`?

Reply:
Obviously non-blocking but in a world/language where a getter can return an optional or null, I think it still makes sense to call it `get_SOMETHING` rather than`maybe_get_SOMETHING`.